### PR TITLE
feature/mx-1649 add preview models and api method

### PR DESCRIPTION
**Pull request by cutoffthetop**
### PR Context
- prep for https://github.com/robert-koch-institut/mex-editor/pull/192
- preview models don't care about required fields and length limits for lists
- they will be used by the editor to be able to search even for potentially "broken" items
- also includes some clean-up of the `mex.common.models` module (imports, exports and docs)

### Added
- add preview models for merged items without cardinality validation
- BREAKING: preview models are now part of all `mex.common.fields` lookups
- add `BackendApiConnector.fetch_preview_items` for fetching previews

### Deprecated
- stop using `ExtractedData`, use `AnyExtractedModel` instead
- stop using `MergedItem`, use `AnyMergedModel` instead
- stop using `AdditiveRule`, use `AnyAdditiveRule` instead
- stop using `SubtractiveRule`, use `AnySubtractiveRule` instead
- stop using `PreventiveRule`, use `AnyPreventiveRule` instead
- stop using `BaseEntity`, use a concrete union instead

### Removed
- removed deprecated `BulkInsertResponse` as alias for `IdentifiersResponse`
- removed unused module export of `mex.common.models.generate_entity_filter_schema`
- removed unused module export of `mex.common.models.generate_mapping_schema`
- drop export `models.ExtractedPrimarySourceIdentifier`, import from `types` instead
- drop export `models.MergedPrimarySourceIdentifier`, import from `types` instead

### CODE REVIEW

File: mex/common/models/base/merged_item.py [hunk 1]
```diff
@@ -3,3 +3,7 @@
 
 class MergedItem(BaseEntity):
     """Base model for all merged item classes."""
+
+
+class PreviewItem(BaseEntity):
```
Comment (rababerladuseladim) [inline mex/common/models/base/merged_item.py:8]: I am surprised to find this in `merged_item.py`. Could this go into it's own `preview_item.py` module? Or rename `merged_item.py` to `items.py`?

**Review by rababerladuseladim — CHANGES_REQUESTED** [review]
Nice, only missing a test for the `BackendConnector.fetch_preview_items` method. Could you add one?

**Comment by cutoffthetop** [comment]
> Nice, only missing a test for the `BackendConnector.fetch_preview_items` method. Could you add one?

done in 37baf24

File: mex/common/models/base/merged_item.py (hunk 1 above)
Comment (cutoffthetop) [reply mex/common/models/base/merged_item.py]: done in c15a4a7 - among some further clean-up

**Review by rababerladuseladim — APPROVED** [review]

**Pull request merged**